sdk/openai/Azure.AI.OpenAI/tests/Samples/Sample07_ChatFunctions.cs

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Azure.Identity;
+using NUnit.Framework;
+
+namespace Azure.AI.OpenAI.Tests.Samples
+{
+    public partial class StreamingChat
+    {
+        [Test]
+        [Ignore("Only verifying that the sample builds")]
+        public async Task ChatFunctions()
+        {
+            string endpoint = "https://myaccount.openai.azure.com/";
+            var client = new OpenAIClient(new Uri(endpoint), new DefaultAzureCredential());
+
+            #region Snippet:ChatFunctions:DefineFunction
+            var getWeatherFuntionDefinition = new FunctionDefinition()
+            {
+                Name = "get_current_weather",
+                Description = "Get the current weather in a given location",
+                Parameters = BinaryData.FromObjectAsJson(
+                new
+                {
+                    Type = "object",
+                    Properties = new
+                    {
+                        Location = new
+                        {
+                            Type = "string",
+                            Description = "The city and state, e.g. San Francisco, CA",
+                        },
+                        Unit = new
+                        {
+                            Type = "string",
+                            Enum = new[] { "celsius", "fahrenheit" },
+                        }
+                    },
+                    Required = new[] { "location" },
+                },
+                new JsonSerializerOptions() {  PropertyNamingPolicy = JsonNamingPolicy.CamelCase }),
```

Comment:
nit: Possibly overthinking this, but I wonder if the Best Practice™️ would be to create the `BinaryData` from a string due to performance, and if our samples should reflect that. Having said that, I do see how using dynamic objects is kinda nice...

Reply:
Good question! I'm not sure about the performance vs. usability implications; that's an explicit topic in the next review sync and we can also follow up with the .NET specific stakeholders. I like this, too, and we of course will have customers doing it (not just via anonymous classes, but also from named classes they use themselves), but if we want to encourage manual JSON manipulation as a standard then I agree we should follow up and align to that. Won't address that here, but great topic for the near future (15 minutes, in fact!).